image/image_linux.go

```diff
@@ -94,10 +94,21 @@ func classicHasSnaps(model *asserts.Model, opts *Options) bool {
 	return model.Gadget() != "" || len(model.RequiredNoEssentialSnaps()) != 0 || len(opts.Snaps) != 0
 }
 
+var newToolingStoreFromModel = NewToolingStoreFromModel
+
 func Prepare(opts *Options) error {
-	model, err := decodeModelAssertion(opts)
-	if err != nil {
-		return err
+	var model *asserts.Model
+	var err error
+	if opts.Classic && opts.ModelFile == "" {
+		// This is needed to support using ubuntu-image to preseed snaps in a rootfs
+		// that has already been built. Since there is no model assertion file
```

Comment:
I'm not quite sure what "already built" means here.  Some rootfses are built with snaps in them already using this. So the explanation needs to be clarified. As I read it naively the comment would make it sounds like this some kind of incremental usage for which setting a model like this doesn't make sense or seems correct if there was already model or snaps initially.